app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,24 @@
+class SessionsController < ApplicationController
+  before_filter :require_user, only: [:destroy]
+  def new
+    redirect_to home_path if current_user
+  end
+
+  def create
+    user = User.find_by(email: params[:email])
+
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      redirect_to home_path, notice: "You are signed in enjoy!"
+    else
+      flash[:error] = "Invalid email or password"
```

Comment:
Make sure you use `flash.now` when rendering. If you don't, the flash message will survive the next request as well, which is not what you expect.

Reply:
Thanks for pointing that out.